drivers/zfs/zfs.go

```diff
@@ -392,12 +392,18 @@ func (d *Driver) Remove(id string) error {
 	name := d.zfsPath(id)
 	dataset := zfs.Dataset{Name: name}
 	err := dataset.Destroy(zfs.DestroyRecursive)
-	if err == nil {
-		d.Lock()
-		delete(d.filesystemsCache, name)
-		d.Unlock()
+	if err != nil {
+		// We must be tolerant in case the image has already been removed,
+		// for example, accidentally by hand.
+		if _, err1 := zfs.GetDataset(name); err1 == nil {
```

Comment:
Also, `zfs.GetDataset` seems more appropriate here than `zfs.Datasets` in my previous commits, since we don't need to search its children recursively.